src/blueprints/tv2/action-factories/tv2-graphics-action-factory.ts

```diff
@@ -150,4 +199,322 @@ export class Tv2GraphicsActionFactory {
+    const downstreamKeyer = this.getDownstreamKeyerMatchingRole(blueprintConfiguration, Tv2DownstreamKeyerRole.FULL_GRAPHICS)
+    const downstreamKeyerEnable: TimelineEnable = { start: 0 }
+    const priority: number = 0
+
+    return [
+      this.videoMixerTimelineObjectFactory.createProgramTimelineObject(sourceInput, enable),
+      this.videoMixerTimelineObjectFactory.createCleanFeedTimelineObject(sourceInput, enable),
+      this.videoMixerTimelineObjectFactory.createLookaheadTimelineObject(sourceInput, enable),
+      this.videoMixerTimelineObjectFactory.createDownstreamKeyerTimelineObject(
+        downstreamKeyer,
+        true,
+        downstreamKeyerEnable,
+        priority
+      ),
+      this.videoMixerTimelineObjectFactory.createUpstreamKeyerTimelineObject(
+        downstreamKeyer,
+        upstreamEnable
+      ),
+      this.audioTimelineObjectFactory.createStudioMicrophonesUpTimelineObject(blueprintConfiguration)
+    ]
+  }
+
+  private createCasparCgFullGraphicsActionFromGraphicsData(blueprintConfiguration: Tv2BlueprintConfiguration, graphicsData: Tv2GraphicsData): Tv2PartAction {
+    if (!blueprintConfiguration.studio.HTMLGraphics) {
+      throw new MisconfigurationException(
+        'Missing configuration of \'HTMLGraphics\' in settings. ' +
+        'Make sure it exists, and contains a value for \'KeepAliveDuration\''
+      )
+    }
+
+    const partId: string = 'fullGraphicsPart'
+    const fullGraphicPiece: PieceInterface = this.createCasparCgFullGraphicsPiece(blueprintConfiguration, graphicsData, partId)
+    return this.createFullGraphicsActionFromGraphicsData(
+      blueprintConfiguration.studio.HTMLGraphics.KeepAliveDuration,
+      partId,
+      fullGraphicPiece,
+      graphicsData
+    )
+  }
+
+  private createCasparCgFullGraphicsPiece(blueprintConfiguration: Tv2BlueprintConfiguration, graphicsData: Tv2GraphicsData, partId: string): PieceInterface {
+    return this.createGraphicsPieceInterface({
+      id: 'fullGraphicPiece',
+      partId,
+      name: graphicsData.name,
+      preRollDuration: blueprintConfiguration.studio.CasparPrerollDuration,
+      pieceLifespan: this.findInfiniteModeFromConfig(blueprintConfiguration, graphicsData),
+      layer: Tv2SourceLayer.PILOT_GRAPHICS,
+      content: this.createCasparCgFullGraphicsPieceContent(blueprintConfiguration, graphicsData),
+      timelineObjects: [
+        this.casparCgGraphicsTimelineObjectFactory.createFullGraphicsTimelineObject(blueprintConfiguration, graphicsData),
+        ...this.createCasparCgFullPilotGraphicsTimelineObjects(blueprintConfiguration)
+      ]
+    })
+  }
+
+  private createCasparCgFullGraphicsPieceContent(blueprintConfiguration: Tv2BlueprintConfiguration, graphicsData: Tv2GraphicsData): Tv2GraphicsContent {
+    const rawGraphicsFolder: string | undefined = blueprintConfiguration.studio.GraphicFolder
+    const graphicsFolder: string = rawGraphicsFolder ? `${rawGraphicsFolder}\\` : ''
```

Comment:
The value in `rawGraphicsFolder` is used further down too.
I could change it so both retrieves it from the `blueprintConfiguration` .
If the original source for `rawGraphicsFolder` were to be changed you would need to update two places then, istead of only one with the current implementation.

Reply:
The only difference between `rawGraphicsFolder` and `graphicsFolder` is a trailing backslash.
That should not make much of a difference when join in `joinAssetToFolder()`. Especially since that method changes all backslahes.
So you should just be able to refactor to my initial suggestion.

Reply:
Done as suggested.